src/app/api/booking/route.ts

```diff
@@ -9,107 +10,87 @@ import { getCurrentUser } from "@/lib/session";
  * Gets game sessions currently available for booking
  */
 
-interface Session {
-  gamesessionId: number;
-  playLevel: string;
-}
-
 export async function POST(request: Request) {
   try {
     const currentUser = await getCurrentUser();
-    if (currentUser === null)
-      return new Response("unauthorized user", { status: 401 });
+    if (!currentUser) return new Response("unauthorized user", { status: 401 });
+
+    const bookingSchema = z.array(
+      z.object({
+        gameSessionId: z.number(),
+        playLevel: z.union([
+          z.literal("beginner"),
+          z.literal("intermediate"),
+          z.literal("advanced"),
+        ]),
+      }),
+    );
 
-    const json = await request.json();
-    const num_of_sessions = json.length;
+    // parse the input array of objects and check if the user has enough sessions
+    const json = bookingSchema.parse(await request.json());
+    const numOfSessions = json.length;
 
-    if (num_of_sessions === 0)
+    if (numOfSessions === 0)
       return new Response("num_of_sessions cannot be equal to 0", {
         status: 400,
       });
 
+    // find user
     const user = await db.query.users.findFirst({
       where: eq(users.id, currentUser!.id),
     });
 
-    if (user?.member === true) {
-      if (num_of_sessions > 2) {
-        return new Response("num_of_sessions cannot be greater than 2", {
-          status: 400,
-        });
-      } else {
-        for (const session of json) {
-          const gameSession = await db.query.gameSessions.findFirst({
-            where: eq(gameSessions.id, session.gamesessionId),
-          });
-
-          const time = new Date();
-          if (time > gameSession!.endTime || time < gameSession!.startTime) {
-            return new Response(null, { status: 400 });
-          }
-        }
-        if (user.remainingSessions < num_of_sessions) {
-          return new Response(null, { status: 400 });
-        }
-
-        await db.transaction(async (tx) => {
-          await tx
-            .update(users)
-            .set({
-              remainingSessions: user.remainingSessions - num_of_sessions,
-            })
-            .where(eq(users.id, currentUser!.id));
-          await tx.insert(bookings).values({
-            userId: currentUser!.id,
-            gameSessionId: json[0].gamesessionId,
-            createdAt: new Date(),
-            difficulty: json[0].playLevel,
-          });
-          await tx.insert(bookings).values({
-            userId: currentUser!.id,
-            gameSessionId: json[1].gamesessionId,
-            createdAt: new Date(),
-            difficulty: json[1].playLevel,
-          });
-        });
-
-        return new Response("success", { status: 204 });
-      }
-    } else {
-      if (num_of_sessions !== 1) {
-        return new Response("num_of_sessions is not equal to 1", {
-          status: 400,
-        });
-      } else {
+    if (numOfSessions <= (user?.member ? 2 : 1) && numOfSessions > 0) {
```

Comment:
better to do an early return to reduce nesting

for example
if (numSessions > maxAllowedSessions) return new Response...

...